(DOCS-2001) Metrics without limits getting started guide (#10673)
DataDog/documentation: content/en/metrics/guide/metrics-without-limits-getting-started.md

* When a count, gauge, or rate metric isn't configured with Metrics without Limits™:

  Datadog can slice, dice, and aggregate the appropriate raw data at query time to provide you mathematically accurate results. 

* When a count, gauge, or rate metric has been configured with Metrics without Limits™ given a specified tag configuration:

  The original raw data must be recombined and aggregated prior to query time to preserve the accuracy of your query results and values. Because of this, Datadog stores six different time/space aggregations (shown below in with check marks) for each remaining tag value combination defined by your tag configuration. Refer to the docs on [the anatomy of a metric][4] to review what time/space aggregations are, and how they work. 
  |           | Time AVG  | Time SUM  | Time MIN  | Time MAX  |
  |-----------|-----------|-----------|-----------|-----------|
  | Space AVG | {{< X >}} | {{< X >}} |           |           |
  | Space SUM | {{< X >}} | {{< X >}} |           |           |
  | Space MIN | {{< X >}} |           |           |           |
  | Space MAX | {{< X >}} |           |           |           |
  

>   If a combination does not have a check mark, it is best to send that metric without using Metrics without Limits™ (keep full cardinality).

Reviewer: ```suggestion
  If a combination you need for query does not have a check mark, it is best to not configure this particular metric with Metrics without Limits™ 
```

Therefore, you can have a resulting indexed custom metric volume greater than an ingested custom metric volume if the tag combination specified does not reduce the number of remaining tag value combinations enough to offset the 6x factor.

Here is a simplified example of how indexed greater than ingested can occur when you start with four tag value combinations on a gauge metric, or in other words, **four custom metrics**:

{{< img src="metrics/guide/before-mwl.jpg" alt="Flow chart with four custom metrics from two hosts"  style="width:80%;" >}}

If you use Metrics without Limits™ and configure on `{endpoint, status}`: 

{{< img src="metrics/guide/after-mwl.jpg" alt="Flow chart with the hosts marked out with x"  style="width:80%;" >}}

Without the host, this leaves three remaining tag combinations:

1. `{endpoint:x, status:200}`
2. `{endpoint:x, status:400}`
3. `{endpoint:y, status:200}`

> However, for each of these three combinations, Metrics without Limits™ stores six pre-aggregated values, so this results in **18 custom metrics total**.

Reviewer: ```suggestion
However, for each of these three combinations, Metrics without Limits™ stores six pre-aggregated values, so this results in **18 custom metrics total**. Therefore, the metric is smaller when left unconfigured.
```

---
title: Getting Started with Metrics without Limits™
kind: guide
is_beta: true
---

<div class="alert alert-warning">This functionality is in private beta.</div>

## Overview

This guide covers how to get started with Metrics without Limits™ and quickly gain the most value from the feature. 

Review the [Metrics without Limits™ FAQ][1].

1. Begin by configuring your Top 20 metrics on the Usage page.

>    You can [configure metrics][2] both in the Metrics Summary UI and with the API.

Reviewer 2: I don't know Metrics very well, but I feel like the bulk of the instructions in this getting started guide are right here in these two lines. It tells me that i should and can do these things, but it's not super clear on how, and the link points to a fairly generic page. How do I find out how to do these things?

Reviewer: Fair - we can include the image already included on the Distribution Metrics' doc / wording from the Customize Tagging section here: https://docs.datadoghq.com/metrics/distributions/#customize-tagging

Author: Woops, I messed up this link, fixed

Author: I think this also sort of ties in with our OKR for the getting started with metrics page, once that's completed we can add links to it from here

>    **Note:** If you're using the Create Tag Configuration API, use the [tag configuration cardinality estimator API][3] first to validate the potential impact of your tag configurations prior to using the API above to create tag configurations. 

Reviewer 2: If they should do this first, then this instruction should come before the step on configuring. Maybe:

1. Review the MwL FAQ.
2. If you are going to use the API, use the estimator
3. Configure your top twenty metrics using the UI or the API.

Reviewer: +

Author: Reordered 👍🏻

   If the UI or the estimator API returns a resulting number of indexed that is [larger than ingested](#why-is-my-indexed-cm-volume-greater-than-ingested-cm-volume), do not save your tag configuration. 

> 2. Use Bulk Metric Configuration to quickly configure tags on multiple metrics.

Reviewer 2: Is this a variation of how to configure tags using the UI? If so, it should be a substep of that step, not a next step.

Reviewer: Could be a substep but is only applicable as a substep if the # of indexed custom metrics is > than the metric's original size of ingested custom metrics.

  The original raw data must be recombined and aggregated prior to query time to preserve the accuracy of your query results and values. Because of this, Datadog stores six different time/space aggregations (shown below in with check marks) for each remaining tag value combination defined by your tag configuration. Refer to the docs on [the anatomy of a metric][6] to review what time/space aggregations are, and how they work. 
  |           | Time AVG  | Time SUM  | Time MIN  | Time MAX  |
  |-----------|-----------|-----------|-----------|-----------|
  | Space AVG | {{< X >}} | {{< X >}} |           |           |
  | Space SUM | {{< X >}} | {{< X >}} |           |           |
  | Space MIN | {{< X >}} |           |           |           |
  | Space MAX | {{< X >}} |           |           |           |
  
  If a combination does not have a check mark, it is best to send that metric without using Metrics without Limits™ (keep full cardinality).

Therefore, you can have a resulting indexed custom metric volume greater than an ingested custom metric volume if the tag combination specified does not reduce the number of remaining tag value combinations enough to offset the 6x factor.

> Here is a simplified example of how indexed greater than ingested can occur when you start with four tag value combinations on a gauge metric, or in other words, **four custom metrics**:

Author: should the second hyphen be between "tag-value" instead of "four-tag"?

> Here is a simplified example of how indexed greater than ingested can occur when you start with four tag value combinations on a gauge metric, or in other words, **four custom metrics**:

Reviewer: to be consistent with our other metrics docs, let's maintain the phrase "tag value combinations" without dashes. but also re:  "indexed-greater-than-ingested" seems awkward to me. why are we requiring dashes here?

> Here is a simplified example of how indexed greater than ingested can occur when you start with four tag value combinations on a gauge metric, or in other words, **four custom metrics**:

Reviewer 2: I am alright with no dashes throughout. ty
